tests/test_collection_length.py

```diff
@@ -28,6 +30,40 @@ class FooFactory(DataclassFactory[Foo]):
     assert len(foo.foo) <= max_val, len(foo.foo)
 
 
+def test_collection_length_with_set_random_collision() -> None:
+    @dataclass
+    class Foo:
+        foo: Set[int]
+
+    class FooFactory(DataclassFactory[Foo]):
+        __model__ = Foo
+        __randomize_collection_length__ = True
+        __min_collection_length__ = 10
+        __max_collection_length__ = 15
+
+    with patch.object(FooFactory.__faker__, "pyint", side_effect=list(range(9)) + list(range(7, 15))):
```

Comment:
Can this be overridden via overriding `get_provider_map` to avoid patching?

Reply:
Your suggestion should work, but since I reverted the code which attempts to generate a unique set, this test is not valid anymore. I've also deleted it.